### ISSUE TITLE
Right.to_some crached with argument error

### ISSUE THREAD

**Issue by borisaka** [dry-rb/dry-monads#25]
``` ruby
M.Right(nil).to_maybe
ArgumentError: nil cannot be some
from /Users/me/.rvm/gems/ruby-2.3.1/gems/dry-monads-0.2.0/lib/dry/monads/maybe.rb:47:in `initialize'
```

I think  `Dry::Monads::Maybe(value)` instead `Maybe::Some.new(value)` is a solution

**Comment by fuadsaud** [comment]
Does it make sense to have a "general" implementation of `to_maybe` for `Either`? I see haskell implements `leftToMaybe` and `rightToMaybe`, which behave a little differently:

http://hackage.haskell.org/package/either-4.3.3.2/docs/Data-Either-Combinators.html#v:leftToMaybe

**Comment by flash-gordon** [comment]
`Either#to_maybe` is equivalent (with minor `nil` exception) to `rightToMaybe`. We don't have an analog of `leftToMaybe` because this means adding one more method to `Either` class and we want to keep it concise. We could add a module with combinators for this implemented as class-level methods 
```ruby
class Either
  module Combinators
    def self.left_to_maybe(either)
      # ...
    end
  end
  C = Combinators
end

Either.C.left_to_maybe
```
Rather verbose, but afraid it's best we can have in ruby

**Comment by fuadsaud** [comment]
@flash-gordon interesting. That could be implemented with refinements as well, in order to keep it more ruby-ish (even though a separate module would keep things even simpler and I like simple), but I definitely understand your point. I'm gonna get a pr to fix this issue since #26 looks stale.

**Comment by flash-gordon** [comment]
OK, this fixed in master
